Option to sort vaults by vault name or cloud

Vladyslav-Soldatenko opened the pull request:
Hi guys. 
This is my proposal for #347 issue. Currently the two ways are supported: alphabetical and by CloudType enum. Since this action is kinda destructive (since the user could have their own sorting via DnD) there is confirmation modal with proper text. We can, of course, add asc/desc option, but I didn't want to complicate things initially as even such sorting is better than none. Feel free to leave comments
<img width="371" height="836" alt="image" src="https://github.com/user-attachments/assets/57f87282-6d17-415b-9d0e-9320d97f4c97" />
<img width="373" height="621" alt="image" src="https://github.com/user-attachments/assets/888a1277-c8a0-42cf-b36e-063b7109dc6e" />

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/cryptomator/android?pullRequest=616) <br/>All committers have signed the CLA.

Vladyslav-Soldatenko committed:
Option to sort vaults by vault name or cloud

JaniruTEC commented:
Hey,
thank you for your PR! :grin: 
I can't say much about the code at this point, but I'd like to put two notes up for discussion:
1) I think it'd be easier (as in less complexity, less dialogs and less destructive operations) to apply the sorting dynamically while populating the list *view.* The user could then pick between location/name/custom-sorting. `Location/Name` each would employ a comparator and disable DnD behavior while `Custom` would retain the current behavior.
2) Building on that we could then switch to a less intrusive component such as menu.

Vladyslav-Soldatenko commented:
Hi, @JaniruTEC. Could you clarify one thing: when user selects "custom", do you want to just unlock DnD or you want to keep previous "custom" sorting saved somewhere when user selects "Location/Name" and restore it when users moves back to custom?
* If we decide NOT to keep previous custom sorting when user moves to non-custom sort, than the operation is quite destructive still (and possible some warning won't hurt).
* If we decide to keep previous custom sorting order, then it raises quite a lot of questions like "what if user has some custom sorting, moved to non-custom, deleted X vaults and added Y, where should they be placed when user moves back to custom order"
